app/screens/channel_notification_preference/channel_notification_preference.js

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {intlShape} from 'react-intl';
+import {
+    Platform,
+    ScrollView,
+    View,
+} from 'react-native';
+
+import {ViewTypes} from '@constants';
+
+import StatusBar from 'app/components/status_bar';
+import SectionItem from 'app/screens/settings/section_item';
+import {preventDoubleTap} from 'app/utils/tap';
+import {alertErrorWithFallback} from 'app/utils/general';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import FormattedText from 'app/components/formatted_text';
+import {t} from '@utils/i18n';
+
+export default class ChannelNotificationPreference extends PureComponent {
+    static propTypes = {
+        actions: PropTypes.shape({
+            updateChannelNotifyProps: PropTypes.func.isRequired,
+        }),
+        channelId: PropTypes.string.isRequired,
+        userId: PropTypes.string.isRequired,
+        notifyProps: PropTypes.object.isRequired,
```

Comment:
this are required here, but can be null in the component that opens this screen, so definitely if any of those are not defined as I mentioned earlier the `NotificationPreference` component should render null.

another nit: you could sort the props alphabetically :)